solr/solr-bridge/src/main/java/org/fao/geonet/solr/SolrHTTPProxy.java

```diff
@@ -0,0 +1,386 @@
+            boolean first = true;        // helps to find the encoding once and only once
+            String s = "";               // piece of file that should contain the encoding
+            while ((len = streamFromServer.read(buf)) > 0) {
+
+                if (first && !isCharsetKnown) {
+                    // charset is unknown try to find it in the file content
+                    for (int i = 0; i < len; i++) {
+                        s += (char) buf[i]; // get the beginning of the file as ASCII
+                    }
+
+                    // s has to be long enough to contain the encoding
+                    if (s.length() > 200) {
+                        String charset = getCharset(s); // extract charset
+
+                        if (charset == null) {
+                            // the charset cannot be found, IE users must be warned
+                            // that the request cannot be fulfilled, nothing good would happen otherwise
+                            if (request.getHeader("User-Agent").toLowerCase().contains("msie")) {
+                                response.sendError(HttpServletResponse.SC_NOT_ACCEPTABLE,
+                                        "Charset of the response is unknown");
+                                streamFromServer.close();
+                                connectionWithFinalHost.disconnect();
+                                streamToClient.close();
+                                return;
+                            }
+                        } else {
+                            // override content-type header and add the charset found
+                            response.addHeader("Content-Type",
+                                    connectionWithFinalHost.getContentType()
+                                            + ";charset=" + charset);
+                            first = false; // we found the encoding, don't try to do it again
+                        }
+                    }
+                }
+
+                // for everyone, the stream is just forwarded to the client
+                streamToClient.write(buf, 0, len);
+            }
+
+            streamFromServer.close();
+            connectionWithFinalHost.disconnect();
+            streamToClient.close();
+        } catch (IOException e) {
+            // connection problem with the host 
+            e.printStackTrace();
+
+            throw new Exception(
+                    String.format("Failed to request Solr at URL %s. " +
+                                    "Check Solr configuration.",
+                            sUrl),
+                    e);
+        }
+    }
+
+    /**
+     * Extract the encoding from a string which is the header node of an xml file
+     *
+     * @param header String that should contain the encoding attribute and its value
+     * @return the charset. null if not found
+     */
+    private String getCharset(String header) {
```

Comment:
To check if to move to some utility class. Same for some additional methods added in this class.